implementations/rust/ockam/ockam_command/src/node/create.rs

```diff
@@ -441,43 +421,49 @@ pub async fn spawn_background_node(
     opts: &CommandGlobalOpts,
     cmd: CreateCommand,
 ) -> miette::Result<()> {
-    let node_name = parse_node_name(&cmd.node_name)?;
-    // Create node state, including the vault and identity if don't exist
-    init_node_state(
-        &opts.state,
-        &node_name,
-        cmd.vault.as_deref(),
-        cmd.identity.as_deref(),
-    )
-    .await?;
-
-    let trust_context_path = match cmd.trust_context_opts.trust_context.clone() {
+    let trust_context = match cmd.trust_context_opts.trust_context.clone() {
         Some(tc) => {
-            let config = opts.state.trust_contexts.read_config_from_path(&tc)?;
-            Some(config.path().unwrap().clone())
+            let trust_context = opts.state.get_trust_context(&tc).await?;
+            Some(trust_context)
         }
         None => None,
     };
 
     // Construct the arguments list and re-execute the ockam
     // CLI in foreground mode to start the newly created node
+    info!("spawing a new node {}", &cmd.node_name);
     spawn_node(
         opts,
-        &node_name,
+        &cmd.node_name,
+        &cmd.identity,
+        &cmd.vault,
         &cmd.tcp_listener_address,
-        cmd.trust_context_opts.project_path.as_ref(),
         cmd.trusted_identities.as_ref(),
         cmd.trusted_identities_file.as_ref(),
         cmd.reload_from_trusted_identities_file.as_ref(),
         cmd.launch_config
             .as_ref()
             .map(|config| serde_json::to_string(config).unwrap()),
-        cmd.authority_identity.as_ref(),
         cmd.credential.as_ref(),
-        trust_context_path.as_ref(),
-        cmd.trust_context_opts.project.as_ref(),
+        trust_context.as_ref(),
+        cmd.trust_context_opts.project_name.clone(),
         cmd.logging_to_file(),
-    )?;
+    )
+    .await?;
 
     Ok(())
 }
+
+async fn check_if_node_is_not_already_running(opts: &CommandGlobalOpts, cmd: &CreateCommand) {
```

Comment:
_Nitpick_

A function prefixed with `check_if` makes me think that it's going to return a bool. I would use the `guard` concept here and rename it to `guard_node_is_not_already_running`

Reply:
Good idea. Done.